packages/maskbook/src/web3/suspends/getERC721TokenDetailed.ts

```diff
@@ -0,0 +1,38 @@
+import type { ERC721 } from '../../contracts/ERC721'
+import { formatChecksumAddress } from '../../plugins/Wallet/formatter'
+import { useERC721TokenContract } from '../contracts/useERC721TokenContract'
+import { useChainId } from '../hooks/useChainState'
+import { useSingleContractMultipleData } from '../hooks/useMulticall'
+import { ERC721TokenDetailed, EthereumTokenType } from '../types'
+
+let token_: ERC721TokenDetailed | undefined = undefined
+async function ERC721TokenDetailed_(address: string, token?: Partial<ERC721TokenDetailed>) {
+    const chainId = useChainId()
+    const erc721TokenContract = useERC721TokenContract(address)
+
+    // compose calls
+    const names = ['name', 'symbol', 'baseURI'] as (keyof ERC721['methods'])[]
+    const callDatas = new Array(3).fill([])
+
+    const [results, calls, _, callback] = useSingleContractMultipleData(erc721TokenContract, names, callDatas)
+
+    await callback(calls)
+
+    // compose
+    const [name, symbol, baseURI] = results.map((x) => (x.error ? undefined : x.value))
+    token_ = {
+        type: EthereumTokenType.ERC721,
+        address: formatChecksumAddress(address),
+        chainId,
+        name: name ?? token?.name ?? '',
+        symbol: symbol ?? token?.symbol ?? '',
+        baseURI: baseURI ?? token?.baseURI ?? '',
+    } as ERC721TokenDetailed
+
+    return
+}
+
+export function getERC721TokenDetailed(address: string, token?: Partial<ERC721TokenDetailed>) {
+    if (!token_) throw ERC721TokenDetailed_(address)
```

Comment:
Suppose we are using this function in different places on the same page. Will this function always return the result of the first call? Cause you are using a local variable to store the fetched token.

Reply:
Token is unchanged and can be used directly in other places